File: code/Summer.java
public class Summer {

  public static int calcSum(int [] nums) {
    int sum = 0;
    for(int num : nums) sum += num;
    return sum;
  }

  public static int[] convertInput(String arg) {
    String[] numStrings = arg.split(",");
    int[] numInts = new int[numStrings.length];
    for(int i = 0; i < numStrings.length; i++) {
      numInts[i] = Integer.parseInt(numStrings[i]);
    }
    return numInts;
  }

  public static void main (String [] args) {
    if(args.length > 0) {
      System.out.println(calcSum(convertInput(args[0])));
    } else {
      System.out.println("Please provide a list of integers in the format 1,2,3");
    }
  }
}

File: code/summer.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

struct num_list {
  int size;
  int *nums;
} num_list;

int calc_sum(struct num_list *nlist) {
  int i, sum = 0;
  for(i = 0; i < nlist->size; i++) sum += nlist->nums[i];
  return sum;
}

struct num_list * convert_input(char *input) {
  struct num_list *nlist = (struct num_list*)malloc(sizeof(struct num_list));
  nlist->nums = (int*)malloc(100);
  int count = 0;
  char *token = strtok(input, ",");
  nlist->nums[count++] = atoi(token);
  while (token = strtok(NULL, ",")) nlist->nums[count++] = atoi(token);
  nlist->size = count;
  return nlist;
}

int main(int argc, char **argv) {
  if(argc > 1) {
    printf("%i\n", calc_sum(convert_input(argv[1])));
  } else {
    fprintf(stderr, "Please provide a list of integers in format 1,2,3\n");
  }
  return 0;
}
